parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -228,34 +183,26 @@ pub mod pallet {
 				Error::<T>::PayloadTooLarge,
 			);
 
-			let next_id = <NextId<T>>::get();
-			if next_id.checked_add(1).is_none() {
-				return Err(Error::<T>::Overflow.into())
-			}
+			let nonce = <Nonce<T>>::get(source_id);
+			let next_nonce = nonce.checked_add(1).ok_or(Error::<T>::Overflow)?;
 
-			<MessageQueue<T>>::try_append(EnqueuedMessage {
-				account: who.clone(),
-				message: Message {
-					id: next_id,
-					target,
-					payload: payload
-						.to_vec()
-						.try_into()
-						.map_err(|_| Error::<T>::PayloadTooLarge)?,
-				},
+			<MessageQueue<T>>::try_append(Message {
```

Comment:
seems  `append` directly should be fine since already checked in previous step?

Reply:
The length validation above isn't encoded in the types, so we have to treat `MessageQueue` as though it's unvalidated. `StorageValue<T>` only has `append` when `T` implements `StorageAppend`, but `BoundedVec` implements `StorageTryAppend` instead.

We could use a phantom type around `BoundedVec` to record the validation in the type, but that won't give us access to `append`. Even if we convert to a mutable `Vec` after validation, that removes the bounds from `BoundedVec`.

We could make our own type like `VecAppendableOnce`, but I'm not sure that's worth it to remove the single `map_err` line below.

Maybe we can remove the `ensure!` above? `try_append` performs the same validation. When the queue is full, we would do some redundant work constructing the `Message` instead of failing early.

Reply:
Gotcha,  not worry it is not a big deal and just keep it is also fine